tools/testing/target_determination/heuristics/random_baseline.py

```diff
@@ -0,0 +1,56 @@
+import random
+from typing import Any, List
+
+from tools.testing.target_determination.heuristics.interface import (
+    HeuristicInterface,
+    TestPrioritizations,
+)
+
+from tools.testing.target_determination.heuristics.utils import get_correlated_tests
+
+# This heuristic should never go into production. It is only used for benchmarking our current heuristics.
+# It randomly assigns tests to the different categories to show the usefulness of our actual heuristics.
+# We should in theory tweak the probabilities to reflect our TTS goal. (Currently 50% of tests should be unranked)
```

Comment:
Mind explaining how would you use this heuristic's metrics as a benchmark?

Reply:
Generally, if you want to measure some statistic, it's important to have a random metric / baseline / control to measure against. 

For example with the stat that we push tests that are likely to fail close to the top in terms of priority, it's pretty easy to just point to this and say our heuristics are doing X% better than random rather than just doing the calculation each time. 

It depends a lot on the metric your measuring but for the first failing test being in the first half of tests run, you'd just compare the how well this baseline does compared to the metric you care about.

To make this a bit better and dry runs in general though you'd have to expose all failing tests during a run one way or another.